_posts/blog/2019-08-14-Monica's-Final.md

```diff
@@ -0,0 +1,75 @@
+
+First, let me bring back an excerpt from my introduction:
+
+> From this summer, I wish to achieve a couple of things: 
+> 1. Learn about the awesome and neat science behind multi-echo fMRI
+> 2. How this library could be advanced in the future, should there be a new publication about denoising ME fMRI data
+> 3. Adding lots and lots of tests to the library so that contributors in the future could extend the library without worries
+> 4. Familiarizing myself with open source, and continue adding onto libraries as a regular contributor!
+
+This is what I wanted to do this summer! It's time to reflect.
+
+### What I wanted to do
+
+I wanted to add a unit test to all the different functions in the tedana library, making it future proof for those who contribute to it in the future. 
+
+My project with `tedana` was to make the testing deck more robust. The project was trying to make contributing more friendly for everyone!
+One big reason being that many who utilize `tedana` in publications are neuroscientists who do not regularly code, but the `tedana` community members want to welcome them to!
+
+This project acheives that goal because it would be a lot more welcoming to have simple unit test functions that not only show them what the functions are doing, but also catch any potential bugs or silly mistakes they could make.
+
+### What I did 
+
+In the end, I added a "smoke test" to all the functions. Smoke tests simply test the different I've recently started on adding a unit test for all the functions.
+
+Links of some of the pull requests and issues! 
+https://github.com/ME-ICA/tedana/pull/313
+https://github.com/ME-ICA/tedana/pull/314
+https://github.com/ME-ICA/tedana/pull/366
+https://github.com/ME-ICA/tedana/pull/367
+https://github.com/ME-ICA/tedana/issues/290
+https://github.com/ME-ICA/tedana/issues/335
+https://github.com/ME-ICA/tedana/issues/376
+
+I don't have a comparison of the before/after of test coverage but a recent screenshot from a recent code coverage graph!
+<figure>
+  <img src="/images/codecoverage.jpg"
+       alt="hometown">
+  <figcaption> Screenshot of code coverage! </figcaption>
+</figure>
+
+### What I didn't do 
+
+Did not acheive all my hope and dreams - adding a unit test to every single function.
+
+The reason was simple, lots in my life happened this summer. Many different times, I wanted to flat out quit GSoC and focus on managing the stress that came from life's challenges. 
+Nonetheless, I'm extremely grateful to Kirstie and the `tedana` members for allowing me to stay onboard for the project, and guiding me along the way!
+
+
+### What was hard
+
+Getting used to the workflow. Open source is extremely independent but also extremely dependent on the members of the community. A big problem I had was wanting to work by myself to understand the library and overcome the different roadblocks I encountered. Many of those would have been solved by the members, yet I was stubborn. 
```

Comment:
This is a great paragraph - I think it's true that this was a big barrier this summer.

Can you add a sentence/paragraph about what you'd do difference, and/or advice to future GSOC students who might feel the same way?

Reply:
I think you also had a lot of difficulty at the start understanding all the complexities of MRI physics and fMRI analyses!! Please do add that in - its absolutely true that `tedana` is a tough project to get started with!